fix[Mentions]: fix overlap of clear and feedback icons in Mentions

QdabuliuQ opened the pull request:
<!--
First of all, thank you for your contribution! 😄
For requesting to pull a new feature or bugfix, please send it from a feature/bugfix branch based on the `master` branch.
Before submitting your pull request, please make sure the checklist below is filled out.
Your pull requests will be merged after one of the collaborators approves.
Thank you!
-->

[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE_CN.md?plain=1)

### 🤔 This is a ...

- [ ] 🆕 New feature
- [x] 🐞 Bug fix
- [ ] 📝 Site / documentation improvement
- [ ] 📽️ Demo improvement
- [ ] 💄 Component style improvement
- [ ] 🤖 TypeScript definition improvement
- [ ] 📦 Bundle size optimization
- [ ] ⚡️ Performance optimization
- [ ] ⭐️ Feature enhancement
- [ ] 🌐 Internationalization
- [ ] 🛠 Refactoring
- [ ] 🎨 Code style optimization
- [ ] ✅ Test Case
- [ ] 🔀 Branch merge
- [ ] ⏩ Workflow
- [ ] ⌨️ Accessibility improvement
- [ ] ❓ Other (about what?)

### 🔗 Related Issues
issue: [https://github.com/ant-design/ant-design/issues/54389](https://github.com/ant-design/ant-design/issues/54389) 

### 💡 Background and Solution
When both suffix-icon and clear-icon appear simultaneously, their positions will overlap.
Check if suffix-icon exists, if it does, add margin: 0 8px to clear-icon.

### 📝 Change Log

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | fix overlap of clear and feedback icons in Mentions |
| 🇨🇳 Chinese | 修复 Mentions 中 clear 和 hasFeedback 重叠问题 |

QdabuliuQ committed:
style[Mentions]: fix overlap of clear and feedback icons in Mentions

QdabuliuQ committed:
fix[Mentions]: refactor icon overlap test in Mentions component

li-jia-nan commented on components/mentions/style/index.ts:
```diff
@@ -152,6 +149,10 @@ const genMentionsStyle: GenerateStyle<MentionsToken> = (token) => {
         },
+
+        [`${componentCls}-suffix:has(> *:nth-child(2)) ${componentCls}-clear-icon`]: {
+          margin: `0 ${unit(8)}`,
```
这里不应该写死吧，`8px` 对应的 token 是 `token.marginXS`

li-jia-nan commented on components/mentions/style/index.ts:
```diff
@@ -152,6 +149,10 @@ const genMentionsStyle: GenerateStyle<MentionsToken> = (token) => {
           },
         },
+
+        [`${componentCls}-suffix:has(> *:nth-child(2)) ${componentCls}-clear-icon`]: {
```
选择器感觉有点奇怪

xiaochena commented:
这样后的新问题：
<img width="1010" height="182" alt="image" src="https://github.com/user-attachments/assets/9eba6214-7e48-4e90-8742-93db862c1c68" />

zombieJ commented on components/mentions/style/index.ts:
```diff
@@ -122,20 +122,17 @@ const genMentionsStyle: GenerateStyle<MentionsToken> = (token) => {
           color: colorTextQuaternary,
           fontSize: fontSizeIcon,
           verticalAlign: -1,
-          // https://github.com/ant-design/ant-design/pull/18151
```
这个注释不用删

QdabuliuQ replied:
好嘞；大佬这个pr也review一下：[https://github.com/ant-design/ant-design/pull/54386](https://github.com/ant-design/ant-design/pull/54386)

QdabuliuQ committed:
style[Mentions]: adjust Mentions component padding and suffix styles

QdabuliuQ commented on components/mentions/__tests__/index.test.tsx:
```diff
@@ -133,4 +134,27 @@ describe('Mentions', () => {
     ).toBe('Yesmeck');
   });
+
+  it('clear icon and feedback icon should not overlap', () => {
```
是在 form 组件里面  custom-feedback-icons demo  加吗～

QdabuliuQ committed:
fix[Mentions]: fix Mentions feedback and clear icon overlap in Form

QdabuliuQ committed:
fix[Mentions]: refactor mentions input padding and margin styles

QdabuliuQ committed:
fix[Mentions]: update Mentions demo snapshots for UI changes

Updated the snapshots for the Mentions component demos to reflect recent UI changes. The basic demo now renders a simpler structure without the Form wrapper and related elements, and warning output has been removed.

li-jia-nan commented on components/mentions/style/index.ts:
```diff
@@ -107,27 +108,26 @@ const genMentionsStyle: GenerateStyle<MentionsToken> = (token) => {
-          paddingInlineEnd: paddingLG,
+          // 当页面中存在 feedback-icon 时，给 clear-icon 添加右边距
+          [`&:has(${antCls}-form-item-feedback-icon) ${componentCls}-clear-icon`]: {
+            marginInlineEnd: unit(token.marginXS),
```
这里不需要 `unit`

QdabuliuQ committed:
style[Mentions]: fix clear-icon margin style in mentions component

Replaces the use of the unit() function with a direct value assignment for marginInlineEnd, ensuring the clear-icon receives the correct margin when a feedback icon is present.

QdabuliuQ committed:
chore[Mentions]: add initial value to Mentions field in form demo

QdabuliuQ committed:
fix[Mentions]: initialValue type in custom-feedback-icons demo

Changed initialValue from array to string for the Mentions field to match expected input type and ensure correct form behavior.

afc163 commented on components/mentions/style/index.ts:
```diff
@@ -97,7 +96,9 @@ const genMentionsStyle: GenerateStyle<MentionsToken> = (token) => {
-        padding: 0,
+        paddingBlock: 0,
+        paddingInlineStart: 0,
+        paddingInlineEnd: unit(token.paddingInline),
```
这里不需要 unit

QdabuliuQ committed:
fix[Mentions]: fix padding style in Mentions component
